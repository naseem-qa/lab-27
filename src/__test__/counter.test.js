import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Counter from '../components/counter/counter.js';

Enzyme.configure({ adapter: new Adapter() });

describe('<Counter />', () => {
    it('can render counter', () => {
        let app = mount(<Counter />);
        let span = app.find('span');
        expect(span.exists()).toBeTruthy();
        expect(app.state('count')).toBe(0);
    });

    it('can render to the DOM', () => {
        const rendered = renderer.create(<Counter />).toJSON();
        expect(rendered).toMatchSnapshot();
    });

    it('can change the counter on click', () => {
        let app = mount(<Counter />);
        let span = app.find('.down');
        span.simulate('click');
        expect(app.state('count')).toBe(-1)
    });
});