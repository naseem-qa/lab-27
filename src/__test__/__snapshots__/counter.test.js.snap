// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> can render to the DOM 1`] = `
<section
  className="counter"
>
  <span
    className="down clicker"
    onClick={[Function]}
  >
     - 
  </span>
  <span
    className="count "
  >
    0
  </span>
  <span
    className="up clicker"
    onClick={[Function]}
  >
     + 
  </span>
</section>
`;
